engine/src/main/java/org/terasology/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -369,15 +365,13 @@ public void send(EntityRef entity, Event event, Component component) {
             return result;
         }
 
-        for (Class<? extends Component> compClass : handlers.keySet()) {
```

Comment:
This method is called very often. I would leave its code as-is, because I think that the simple iteration is more performant than the conversion and lambda-based filtering.